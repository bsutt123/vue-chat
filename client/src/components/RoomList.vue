<template>
  <div :class='$style.container'>
    <div :class='$style.signInBox'>
      <sign-in></sign-in>
      <sign-up></sign-up>
    </div>
    <div :class='$style.headingBox'>
      <h2 :class='$style.heading'> Rooms </h2>
    </div>
    <ul :class='$style.roomList'>
      <li
        v-for='(room, index) in rooms'
        v-bind:key='index'
        :class='$style.roomItem'
        @click="setActiveRoom(room)"
      >
        {{ room.name }}
      </li>
    </ul>
    <room-modal></room-modal>
  </div>
</template>

<script>
import { createNamespacedHelpers } from 'vuex';
import SignIn from './SignIn';
import SignUp from './SignUp';
import RoomModal from './RoomModal';

const { mapState, mapActions } = createNamespacedHelpers('rooms');

export default {
  name: 'RoomList',
  computed: {
    ...mapState({
      rooms: state => state.rooms,
      activeRoom: state => state.activeRoom,
    }),
  },
  methods: {
    ...mapActions([
      'getRooms',
      'setActiveRoom',
    ]),
  },
  created() {
    this.getRooms();
  },
  components: {
    'sign-in': SignIn,
    'sign-up': SignUp,
    'room-modal': RoomModal,
  },
};
</script>

<style lang="postcss" module>
@import '../styles/colors.css';

.container {
  padding: 1rem;
  height: 100%;
  background-color: var(--colorPrimary);
  color: white;
}

.signInBox {
  text-align: center;

  & > *:not(:last-child) {
    margin-right: 1rem;
  }
}

.headingBox {
  display: grid;
  grid-template-columns: repeat(autofit, minmax(auto, 1fr));
  text-align: center;
}

.heading {
  margin-bottom: 0.5rem;
}

.newRoomButton {
  margin-bottom: 1rem;
  padding: .5rem 1rem;
  color: black;
  transition: all .1s linear;

  &:hover {
    background-color: gray(210);
    transform: translateY(-4px);
  }
}

.roomList {
  font-size: .9rem;
  list-style: none;
  margin-bottom: 1rem;
  max-width: 200px;

  & > *:not(:last-child) {
    margin-bottom: 0.5rem;
  }
}

.roomItem {
  cursor: pointer;
  transition: all .1s linear;

  &:hover {
    transform: translateY(-2px) scale(1.125);
  }
}


</style>
