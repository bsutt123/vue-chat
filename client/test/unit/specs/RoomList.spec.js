import { shallow, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VModal from 'vue-js-modal';

import { CssModuleTestHelperMixin } from '../utils/test-helper';
import RoomList from '../../../src/components/RoomList';
import storeObject from '../utils/testStore';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VModal);
localVue.use(CssModuleTestHelperMixin);

describe('RoomList', () => {
  let store;
  let wrapper;
  let actions;

  beforeEach(() => {
    store = new Vuex.Store(storeObject);
    wrapper = shallow(RoomList, {store, localVue})
    actions = storeObject.modules.rooms.actions
  });

  it('calls the store action when expected', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders the correct number of rooms', () => {
    expect(wrapper.find('.roomList').findAll('.roomItem').length).toEqual(3)
  });

  it('calls setActiveRoom on click', () => {
    const roomItems = wrapper.findAll('.roomItem')
    roomItems.trigger('click')
    expect(actions.setActiveRoom.mock.calls.length).toEqual(3)
  })

});
