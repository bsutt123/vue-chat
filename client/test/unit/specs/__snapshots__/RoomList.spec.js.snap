// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomList calls the store action when expected 1`] = `
<div class="container">
    <div class="signInBox">
        <!---->
        <!---->
    </div>
    <div class="headingBox">
        <h2 class="heading"> Rooms </h2>
    </div>
    <ul class="roomList">
        <li class="roomItem">
            Room 1
        </li>
        <li class="roomItem">
            Room 2
        </li>
        <li class="roomItem">
            Room 3
        </li>
    </ul>
    <!---->
</div>
`;
